### PR TITLE
SB-24639 feat : Observation Offline

### PR DESCRIPTION
**Pull request by vishwanath1004**

### CODE REVIEW

File: src/app/download-manager/download-manager.page.html [hunk 1]
```diff
@@ -4,8 +4,11 @@
 
     <app-no-downloads *ngIf="!downloadedContents.length"></app-no-downloads>
 
-    <app-downloads-tab #downloadsTab *ngIf="downloadedContents.length" [downloadedContents]="downloadedContents"
```
Comment (kiranharidas187) [inline src/app/download-manager/download-manager.page.html:7]: Why is this condition removed?

File: src/app/download-manager/download-manager.page.ts [hunk 2]
```diff
@@ -169,13 +172,24 @@ export class DownloadManagerPage implements DownloadManagerPageInterface, OnInit
               })
           }
         }
-
+        await this.storage
```
Comment (kiranharidas187) [inline src/app/download-manager/download-manager.page.ts:175]: check if any unit test cases are failing

File: src/app/home/admin-home/admin-home.page.ts [hunk 3]
```diff
@@ -82,8 +82,11 @@ export class AdminHomePage implements OnInit, OnDestroy, OnTabViewWillEnter {
     private qrScanner: SunbirdQRScanner,
     private storage: LocalStorageService,
     private unnatiService: UnnatiDataService,
-    private db: DbService
-  ) {}
+    private db: DbService,
+    private utilsService: UtilsService
```
Comment (kiranharidas187) [inline src/app/home/admin-home/admin-home.page.ts:86]: Why is this imported. Revert all the changes in this file if not required

File: src/app/manage-learn/observation/observation.module.ts [hunk 4]
```diff
@@ -24,6 +24,6 @@ import { CoreModule } from '../core/core.module';
     IonicModule,
     FormsModule,
   ],
-  providers: [ObservationService],
+  providers: [],
```
Comment (kiranharidas187) [inline src/app/manage-learn/observation/observation.module.ts:27]: rm if not required

File: src/app/manage-learn/shared/components/item-list-card/item-list-card.component.scss [hunk 5]
```diff
@@ -8,6 +8,16 @@
     color: #{$blue};
 }
 
+
+.ellipsis{
+    text-overflow: ellipsis;
+    overflow: hidden;
+    width: 166px;
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/item-list-card/item-list-card.component.scss:15]: Why is width hardcoded

File: src/app/download-manager/downloads-tab/downloads-tab.component.ts [hunk 6]
```diff
@@ -326,4 +331,46 @@ export class DownloadsTabComponent implements OnInit {
...
+    }).then(success => {
+      this.obsService.obsTraceObj.programId = programId;
+      this.obsService.obsTraceObj.solutionId = solutionId;
+      this.obsService.obsTraceObj.name = solutionName;
+      this.obsService.obsTraceObj.programName = solution.programName;
+      this.obsService.obsTraceObj.observationId = solution.observationId;
+    });
+ }
+
+ async showObsDeletePopup(identifier){
```
Comment (rejneesh1) [inline src/app/download-manager/downloads-tab/downloads-tab.component.ts:353]: @vishwanath1004  do we need this alert box seperately, there is already one alert box by default on click of delete from downloads page

**Pull request merged**